PowerPointLabs/PowerPointLabs/CropLab/CropToSame.cs

```diff
@@ -12,22 +16,25 @@ public class CropToSame
         public static bool CropSelection(PowerPoint.ShapeRange shapeRange)
         {
             bool hasChange = false;
-            var refShape = shapeRange[1];
-            float refScaleWidth = Graphics.GetScaleWidth(refShape);
-            float refScaleHeight = Graphics.GetScaleHeight(refShape);
 
-            float cropTop = Math.Max(refShape.PictureFormat.CropTop, Epsilon);
-            float cropBottom = Math.Max(refShape.PictureFormat.CropBottom, Epsilon);
-            float cropLeft = Math.Max(refShape.PictureFormat.CropLeft, Epsilon);
-            float cropRight = Math.Max(refShape.PictureFormat.CropRight, Epsilon);
+            Shape refObj = shapeRange[1];
+            bool isRefObjShape = Graphics.IsShape(refObj);
+
+            float refScaleWidth = Graphics.GetScaleWidth(refObj);
+            float refScaleHeight = Graphics.GetScaleHeight(refObj);
+
+            float cropTop = isRefObjShape ? Epsilon : Math.Max(refObj.PictureFormat.CropTop, Epsilon);
+            float cropBottom = isRefObjShape ? Epsilon : Math.Max(refObj.PictureFormat.CropBottom, Epsilon);
+            float cropLeft = isRefObjShape ? Epsilon : Math.Max(refObj.PictureFormat.CropLeft, Epsilon);
+            float cropRight = isRefObjShape ? Epsilon : Math.Max(refObj.PictureFormat.CropRight, Epsilon);
```

Comment:
You should use an if statement. Doing it like this will end up checking the same thing 4 times.